### ISSUE TITLE
Chart: 8-color palette insufficient for multi-series and multi-run comparison

### ISSUE THREAD

**Issue by ryanoneill** [ryanoneill/envision#365]
## Summary

The color palette for chart series is limited to the 8 basic ANSI terminal colors (Red, Blue, Green, Cyan, Yellow, Magenta, White, and their dark variants). When rendering more than 4-5 series, colors become difficult to distinguish — and beyond 8, they must repeat.

## Why this matters

Common ML visualization scenarios that need many distinguishable series:

- **Multi-run experiment comparison:** Comparing 8-15 training runs with different hyperparameters is routine in experiment tracking (W&B, MLflow). Each run needs a distinct color.
- **Per-layer metrics:** A transformer with 12+ layers, each plotted as a separate series for gradient norms, activation stats, or probing accuracy.
- **Per-class performance:** Datasets with 10-20 classes, each needing a distinct color in confusion-related scatter plots or per-class curves.
- **Ensemble member tracking:** 5-10 ensemble members with their individual predictions.

With only 8 basic colors, you run out fast. Blue and Cyan look similar in many terminals. Red and Magenta overlap. The effective distinguishable palette is more like 5-6 colors.

## Suggested approach

Modern terminals (iTerm2, Alacritty, Kitty, Windows Terminal, WezTerm) support full 24-bit RGB color. Ratatui supports `Color::Rgb(r, g, b)`.

1. **Define a curated 16-20 color palette** optimized for distinguishability on dark terminal backgrounds. Base it on established palettes like Tableau 10/20, ColorBrewer, or D3's categorical scales.

2. **Auto-assign from the extended palette** when series don't have explicit colors set. The first series gets color 0, second gets color 1, etc., cycling through the full palette.

3. **Keep basic ANSI colors as fallback** for terminals that don't support RGB (rare in 2026, but good practice).

Example palette (dark-background optimized):
```rust
const PALETTE: &[Color] = &[
    Color::Rgb(31, 119, 180),   // muted blue
    Color::Rgb(255, 127, 14),   // orange
    Color::Rgb(44, 160, 44),    // green
    Color::Rgb(214, 39, 40),    // red
    Color::Rgb(148, 103, 189),  // purple
    Color::Rgb(140, 86, 75),    // brown
    Color::Rgb(227, 119, 194),  // pink
    Color::Rgb(127, 127, 127),  // gray
    Color::Rgb(188, 189, 34),   // olive
    Color::Rgb(23, 190, 207),   // teal
    // ... extend to 16-20
];
```
